react-carousel/my-app/src/Carousel.tsx

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from 'react';
+import { Image } from './Image';
+import { GrPrevious } from 'react-icons/gr';
+import { GrNext } from 'react-icons/gr';
+import { FaRegCircle } from 'react-icons/fa';
+type Image = {
+  src: string;
+  alt: string;
+};
+type Props = {
+  images: Image[];
+};
+export function Carousel({ images }: Props) {
+  const [index, setIndex] = useState(0);
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      setIndex((index + 1) % images.length);
+    }, 3000);
+    return () => {
+      clearInterval(intervalId);
+    };
+  }, [index]);
+  return (
+    <div className="row">
```

Comment:
You put almost all the code in one component. I was hoping you'd get practice in this challenge coordinating between several different components.